packages/flutter/lib/src/painting/binding.dart

```diff
@@ -70,8 +70,25 @@ mixin PaintingBinding on BindingBase, ServicesBinding {
   ImageCache createImageCache() => ImageCache();
 
   /// Calls through to [dart:ui] with [decodedCacheRatioCap] from [ImageCache].
-  Future<ui.Codec> instantiateImageCodec(Uint8List list) {
-    return ui.instantiateImageCodec(list);
+  ///
+  /// The [cacheWidth] and [cacheHeight] parameters, when specified, indicate the
+  /// size to decode the image to.
+  ///
+  /// Both [cacheWidth] and [cacheHeight] must be positive values greater than or
+  /// equal to 1. It is valid to specify only one of [cacheWidth] and
+  /// [cacheHeight] with the other remaining null, in which case the omitted
+  /// dimension will decode to its original size.
+  Future<ui.Codec> instantiateImageCodec(Uint8List bytes, {
+    int cacheWidth,
+    int cacheHeight,
+  }) {
+    assert(cacheWidth == null || cacheWidth > 0);
+    assert(cacheHeight == null || cacheHeight > 0);
+    return ui.instantiateImageCodec(
+      bytes,
+      targetWidth: cacheWidth,
+      targetHeight: cacheHeight,
+    );
```

Comment:
is it ok for both to be null? docs say no, asserts say yes

Reply:
Updated docs.